### PR TITLE
User story 32 33 34 - Merchant Order Show Page,  Fulfillment, and Sad Path

### PR DESCRIPTION
**Pull request by MackHalliday**
- Merchant user can view order show page and see information pertaining to them. (item_orders pertaining to them)

- Merchant user cannot see item_orders pertaining to other merchants

- If merchant user has no item_orders under the order, merchant user redirected to 404 page

- Merchant user can fulfill item_order by click on fulfill item button. Merchant user does not see this button if they do not have enough items to fulfill the order

- Sad path included. Merchant user does not see button to fulfill order if user's desire quantity is greater than merchant's inventory. 
 
- New route created with merchant order show page and patch route for merchant order item_order

- Added additional orders and items to seed file 

Tested Live
Rspec OK
SimpleCov OK

### CODE REVIEW

File: app/models/merchant_user.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
-class MerchantUser < ApplicationRecord
-  belongs_to :merchant
-  belongs_to :user 
-end
+# TO BE DELETED
+# class MerchantUser < ApplicationRecord
+#   belongs_to :merchant
+#   belongs_to :user
+# end
```
Comment (leiyakenney) [inline app/models/merchant_user.rb]: I think moving forward if we agree as a group the file needs to be deleted, we can all just delete that file in our individual code before pushing up. Not that it's a big deal, but I have the feeling if we all do this, we'll end up with a bunch of files with just commented out code :)

File: app/models/item.rb [hunk 2]
```diff
@@ -24,4 +24,16 @@ def sorted_reviews(limit, order)
...
+    self.update(inventory: self.inventory - item_order.quantity)
+  end
+
+  def cannot_fulfill_message
+    if self.inventory > 0
+      "Cannot fulfill. Only #{self.inventory} remaining."
+    else
+      "Cannot fulfill. There are no #{self.name} items remaining."
+    end
+  end
```
Comment (leiyakenney) [inline app/models/item.rb]: I like the use of class methods here!

File: app/views/merchant/orders/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+              <td><%= "Fulfilled" %></td>
+            <% end %>
+          <% else %>
+            <td> <%= item_order.item.cannot_fulfill_message %></td>
+        <% end %>
+        <td><%= item_order.item.inventory%></td>
+      </tr>
+    </section>
+  <% end %>
+</table>
```
Comment (leiyakenney) [inline app/views/merchant/orders/show.html.erb]: Looks good to me; I like the sad paths you accounted for here.

File: db/seeds.rb [hunk 4]
```diff
@@ -55,3 +64,23 @@
               email: "recoffice@email.com",
               password: "Waffles",
               role: 3)
+
+order_1 = Order.create!(name: "Beth", address: "123 Happy St", city: "Denver", state: "CO", zip: "80205")
```
Comment (nancylee713) [inline db/seeds.rb]: Thanks for supplying the seed file with sample data!

File: spec/models/item_orders_spec.rb [hunk 5]
```diff
@@ -33,20 +33,44 @@
...
     it 'grandtotal_per_order' do
       expect(ItemOrder.grandtotal_per_order(@order_1.id)).to eq(100)
     end
+
+    it 'can return true if status is pending' do
+      expect(@item_order_1.pending?).to eq(true)
+      expect(@item_order_2.pending?).to eq(false)
+    end
+
+    it 'can return true if item is in stock' do
```
Comment (nancylee713) [inline spec/models/item_orders_spec.rb]: Did you mean to test #instock? method here?

File: spec/features/users/merchant_users/orders/show_spec.rb [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    @order_2 = create(:order)
+      @item_order_4 = @regular_user_1.item_orders.create(order: @order_2, item: @item_2, quantity: 100, price: @item_2.price, user: @regular_user_1, status: "pending")
+
+    @order_3 = create(:order)
+      @item_order_5 = @regular_user_1.item_orders.create(order: @order_3, item: @item_4, quantity: 18, price: @item_4.price, user: @regular_user_1, status: "pending")
+
+    @order_4 = create(:order)
+      @item_order_6 = @regular_user_1.item_orders.create(order: @order_4, item: @item_5, quantity: 15, price: @item_5.price, user: @regular_user_1, status: "pending")
+
+    @merchant_admin_1 = create(:user, role: 1, merchant: @merchant_shop_1)
```
Comment (nancylee713) [inline spec/features/users/merchant_users/orders/show_spec.rb]: merchant_admin's role should be 2, and merchant_employee is 1 ?

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -102,4 +101,27 @@
...
+        item_2 = merchant_shop_1.items.create!(attributes_for(:item, name: "Item 2"))
+
+      order_1 = create(:order)
+        item_order_1 = regular_user_1.item_orders.create!(order: order_1, item: item_1, quantity: 2, price: item_1.price, user: regular_user_1)
+        item_order_2 = regular_user_1.item_orders.create!(order: order_1, item: item_2, quantity: 8, price: item_2.price, user: regular_user_1)
+
+      order_2 = create(:order)
+        item_order_4 = regular_user_1.item_orders.create(order: order_2, item: item_2, quantity: 18, price: item_2.price, user: regular_user_1)
+
+      merchant_admin = create(:user, role: 1, merchant: merchant_shop_1)
```
Comment (nancylee713) [inline spec/models/user_spec.rb]: Here too, do we need to have a merchant_admin (with role 2) or merchant_employee (with role 1) ?

File: app/models/merchant_user.rb (hunk 1 above)
Comment (tschaffer1618) [reply app/models/merchant_user.rb]: I think this is one of the reasons that this merge request has "big" (aka not able to be resolved on github) conflicts to resolve.

File: spec/models/user_spec.rb (hunk 7 above)
Comment (MackHalliday) [reply spec/models/user_spec.rb]: Updated and pushed up! Thanks

**Comment by MackHalliday** [comment]
Updating to reflect recent db changes

**Pull request closed without merging**